sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala

```diff
@@ -456,8 +456,17 @@ object JdbcUtils extends Logging {
 
         case StringType =>
           (array: Object) =>
-            array.asInstanceOf[Array[java.lang.String]]
-              .map(UTF8String.fromString)
+            array match {
```

Comment:
looking at how we handle StringType, we call `ResultSet.getString` even if the underling type is not string, but uuid, inet, cidr, etc. I think we shoud follow the behavior here, i.e. just call `array.asInstanceOf[Array[java.lang.Object]].map(obj => UTF8String.fromString(obj.toString))`